token-swap/js/scripts/typegen.sh

```diff
@@ -0,0 +1,19 @@
+set -e
+
+# Generate typescript declarations
+npx tsc -p tsconfig.d.json -d
+
+# Flatten typescript declarations
+npx rollup -c rollup.config.types.js
+
+# Replace export with closing brace for module declaration
+sed -i '$s/export {.*};/}/' lib/index.d.ts
```

Comment:
Really we need some scripting to end up with functional module? This comes from web3.js.
I am no js expert so I followed

Reply:
I'm not too familiar with this either, @jstarry is this an ok model to follow for other JS bindings?

Reply:
Done

Reply:
Probably not needed. We needed to do this for `@solana/web3.js` because we needed to make sure that transitive dependency on `@solana/web3.js` with an old `module.d.ts` wouldn't override the typings included in the new version. Because of TypeScript's module resolution algorithm, any types declared in a module will take precedence. I think it's unlikely that `@solana/spl-token-swap` will be used as both a high-level dependency as well as a transitive dependency so I think it's safe to skip this step

Reply:
I removed the sed magic, however, not sure which part exactly isn't needed.

Type generation is now reduced to:
https://github.com/solana-labs/solana-program-library/blob/9a61839f63786c7c7bb84b43cf7b6300dae7dfd5/token-swap/js/package.json#L41